docs/wfforward.md

```diff
@@ -0,0 +1,1075 @@
+
+In order to make that happen, we'll 'cap' our breakpoints with 0 and 1:
+
+
+```python
+brk.append(1.0)
+brk.insert(0,0.0)
+print(brk)
+```
+
+    [0.0, 0.25, 0.75, 1.0]
+
+
+Now, we can zip everything together to generate our new edges:
+
+
+```python
+pg=list([*tuple(i for i in pnodes)]*int(len(brk)/2))
+print('Parent nodes are:',pg)
+print('Left end of segments are',brk)
+for l,r,p in zip(brk[:len(brk)-1],brk[1:len(brk)],pg):
+    print('left =',l,'right =',r,'parent =',p)
+```
+
+    Parent nodes are: [163, 162, 163, 162]
+    Left end of segments are [0.0, 0.25, 0.75, 1.0]
+    left = 0.0 right = 0.25 parent = 163
+    left = 0.25 right = 0.75 parent = 162
+    left = 0.75 right = 1.0 parent = 163
+
+
+If, however, our breakpoints contained a zero:
+
+
+```python
+brk = [0,0.25,0.75]
+```
+
+Our verbal description becomes:
+
+* $[0,0.25)$ passed from 162 to child
+* $[0.25,0.75)$ passed from 163 to child
+* $[0.75,1)$ passed from 162 to child
+
+To make this work, we simply need to reverse the input parent node IDs and run the same machinery:
+
+
+```python
+brk.append(1.0)
+pg=list([*tuple(i for i in reversed(pnodes))]*int(len(brk)/2))
+print('Parent nodes are:',pg)
+print('Left end of segments are',brk)
+for l,r,p in zip(brk[:len(brk)-1],brk[1:len(brk)],pg):
+    print('left =',l,'right =',r,'parent =',p)
+```
+
+    Parent nodes are: [162, 163, 162, 163]
+    Left end of segments are [0, 0.25, 0.75, 1.0]
+    left = 0 right = 0.25 parent = 162
+    left = 0.25 right = 0.75 parent = 163
+    left = 0.75 right = 1.0 parent = 162
+
+
+**End of gory details**
+
+
+```python
+def simplify_nodes_edges_mutations(nodes,edges,sites,mutations,temp_nodes,temp_mutations,dt):
+    """
+    :param nodes (msprime.NodeTable): All nodes simplified so far
+    :param edges (msprime.EdgeTable): All edges simplified so far
+    :param temp_nodes (msprime.NodeTable): All new nodes since the last simplification happened.
+    :param dt (int): The number of generations since the last simplification
+    """
+    
+    # Our first sets of operations are what we've used in 
+    # previous examples
+    if len(edges) == 0:
+        return nodes,edges,sites,mutations
+    
+    nodes.set_columns(time=nodes.time+float(dt),flags=nodes.flags)
+    
+    t = temp_nodes.time
+    t -= t.max()
+    t *= -1.0
+ 
+    nodes.append_columns(time=t,flags=np.ones(len(temp_nodes),dtype=np.uint32))
+    samples = np.where(nodes.time == 0.0)[0]
+    
+    # What we need to do now is add update our site and mutation tables
+    
+    # First, convert our meta data into raw bytes, which is handled via 
+    # Python's built-in pickling:
+    encoded, offset = msprime.pack_bytes(list(map(pickle.dumps,[i[1] for i in temp_mutations])))
+    
+    sites.append_columns(position=[i[1].pos for i in temp_mutations],
+                         # encode ancestral state as '0'
+                        ancestral_state=np.zeros(len(temp_mutations), np.int8) + ord('0'),
```

Comment:
This will be mysterious to people I think. It would be nice if we could avoid doing this and just record the sites/mutations directly into the main table with ``add_row``, as this avoids all the messy details of encoding strings.